### PR TITLE
Tests for the API and other changes

### PR DESCRIPTION
**Pull request by IceWreck**
- [x] Modify API to give build status for every chroot
- [x] Tests for the API (partially done)
- [x] replace/supplement existing flexmock test for the logs view with actual/db based test

### CODE REVIEW

File: packit_service/service/api/copr_builds.py [hunk 1]
```diff
@@ -60,7 +60,8 @@ def get(self):
                     "project": build.project_name,
                     "owner": build.owner,
                     "build_id": build.build_id,
-                    "status": build.status,
+                    "status": build.status,  # Legacy, remove later.
+                    "status_per_chroot": {},
```
Comment (TomasTomecek) [inline packit_service/service/api/copr_builds.py:64]: I wouldn't personally mind breaking the API here and just set the dict to status

others may have a different opinion

Comment (IceWreck) [reply packit_service/service/api/copr_builds.py:64]: I'll just build tests for the entire API. It was long overdue anyways #489

**Comment by IceWreck** [comment]
I wanted to use pytests's (official) flask plugin but it isn't in dnf so I had change pytest's source from dnf to pip. (pytest from dnf but pytest-flask from pip don't work together)

Now the API tests are mostly working but an openshift integration test broke.

**Comment by TomasTomecek** [comment]
I'd say the fedpkg test data need to be just regenerated.

**Comment by IceWreck** [comment]
> I'd say the fedpkg test data need to be just regenerated.

Worked. Thanks.

File: packit_service/service/api/copr_builds.py (hunk 1 above)
Comment (IceWreck) [reply packit_service/service/api/copr_builds.py:64]: > I wouldn't personally mind breaking the API here and just set the dict to status
> 
> others may have a different opinion

So should I remove status and chroots, both are redundant if status_per_chroot exists.

Comment (TomasTomecek) [reply packit_service/service/api/copr_builds.py:64]: since no one from the team raised an opinion here, let's keep it as it is now

**Pull request merged**